rDNA/R/rDNA.R

```diff
@@ -1806,4 +1807,289 @@ print.dna_barplot <- function(x, trim = 30, attr = TRUE, ...) {
+#'   When set to \code{0.5}, space between two bars is the same as thickness of
+#'   bars.
+#' @param axisWidth Thickness of the x-axis which separates agreement from
+#'   disagreement.
+#' @param truncate Sets the number of characters to which axis labels should be truncated.
+#' @param exclude.min Reduces the plot to entities with a minimum frequency of statements.
+#'
+#' @examples
+#' \dontrun{
+#' dna_init()
+#' dna_sample()
+#'
+#' dna_openDatabase("sample.dna", coderId = 1, coderPassword = "sample")
+#' 
+#' # compute barplot data
+#' b <- dna_barplot(statementType = "DNA Statement",
+#'                  variable = "concept",
+#'                  qualifier = "agreement")
+#' 
+#' # plot barplot with ggplot2                
+#' autoplot(b)
+#' }
+#'
+#' @author Johannes B. Gruber, Tim Henrichsen
+#'
+#' @seealso \link{dna_barplot}
+#' 
+#' @importFrom ggplot2 autoplot
+#' @importFrom ggplot2 ggplot
+#' @importFrom ggplot2 aes_string
+#' @importFrom ggplot2 geom_line
+#' @importFrom ggplot2 theme_minimal
+#' @importFrom ggplot2 theme
+#' @importFrom ggplot2 geom_bar
+#' @importFrom ggplot2 position_stack
+#' @importFrom ggplot2 coord_flip
+#' @importFrom ggplot2 element_blank
+#' @importFrom ggplot2 element_text
+#' @importFrom ggplot2 scale_color_identity
+#' @importFrom ggplot2 scale_fill_identity
+#' @importFrom ggplot2 geom_text
+#' @importFrom ggplot2 .pt
+#' @importFrom ggplot2 annotate
+#' @importFrom ggplot2 scale_x_discrete
+#' @export
+autoplot.dna_barplot <- function(object, 
+                                 lab.pos = "Agreement",
+                                 lab.neg = "Disagreement",
+                                 lab = TRUE,
+                                 colors = FALSE,
+                                 fontSize = 12,
+                                 barWidth = 0.6,
+                                 axisWidth = 1.5,
+                                 truncate = 40,
+                                 exclude.min = NULL) {
+  
+  # Get qualifier values
+  w <- attr(object, "intValues")
+  
+  # Check if qualifier is binary
+  binary <- all(w %in% c(0, 1))
+  
+  # Compute total values per entity
+  object$sum <- rowSums(object[, colnames(object) %in% w])
+  
+  # Exclude minimum number of statements per entity
+  if (!is.null(exclude.min)) {
+    object <- object[object$sum >= exclude.min,]
+  }
+  
+  # Stack agreement and disagreement
+  object2 <- cbind(object$Entity, utils::stack(object, select = colnames(object) %in% w))
+  colnames(object2) <- c("entity", "frequency", "agreement")
+  
+  object <- object[order(object$sum, decreasing = TRUE),]
+  
+  object2$entity <- factor(object2$entity, levels = rev(object$Entity))
+  
+  # Get colors
+  if (colors) {
+    object2$color <- object$Color[match(object2$entity, object$Entity)]
+    object2$text_color <- "black"
+      object2$text_color[sum(grDevices::col2rgb(object2$color) * c(299, 587, 114)) / 1000 < 123] <- "white"
```

Comment:
Add comment with an explanation about what is happening here. Also, fix indentation.